examples/future_execution.rb

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env ruby
+
+require_relative 'example_helper'
+
+class CustomPassedObject
+  attr_reader :id, :name
+
+  def initialize(id, name)
+    @id = id
+    @name = name
+  end
+end
+
+class CustomPassedObjectSerializer < ::Dynflow::Serializers::Abstract
+  def serialize
+    object = @args.first
+    # Serialized output can be anything that is representable as JSON: Array, Hash...
+    { :id => object.id, :name => object.name }
+  end
+
+  def deserialize(args)
```

Comment:
Why? When the serializer is created it gets the arguments it should pass to the action.  It stores them as they are, they are not serialized yet. The serialization itself happens in this method, using the 'raw' arguments.